src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const keyValue = item[key] as unknown as string;
+
+    if (!acc[keyValue]) {
+      acc[keyValue] = [];
+    }
+
+    acc[keyValue].push(item);
+
+    return acc;
+  }, {} as GroupsMap<T>);
```

Comment:
The reducer function implicitly assumes that all keys are strings. If `T` has keys that are not strings, this could lead to unexpected behavior. Ensure that the key provided can always be safely treated as a string, or adjust the logic to handle different key types.